### PR TITLE
build: CI using Github Actions and bump dependencies

### PR DESCRIPTION
**Pull request by jtmal-signalfx**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+permissions: read-all
+
+jobs:
+  unit-tests:
+    runs-on: ubuntu-latest
+    permissions: read-all
+    strategy:
+      fail-fast: false
+      matrix:
+        # node 8 is technically supported but it is quite difficult to get it to build correctly these days
```
Comment (Rauno56) [inline .github/workflows/ci.yml:16]: What's the issue with v8? What about v12+?

File: package.json [hunk 2]
```diff
@@ -3,29 +3,46 @@
...
+    "build": "node-pre-gyp build",
+    "install": "node-pre-gyp install --fallback-to-build",
+    "lint:fix": "eslint src --fix",
     "lint": "eslint src",
-    "test": "mocha"
+    "pretest": "npm run build",
+    "test": "jest",
+    "test:watch": "jest --watch"
+  },
+  "binary": {
```
Comment (Rauno56) [inline package.json:19]: What are those for? Distribution?

Comment (jtmal-signalfx) [reply package.json:19]: `node-pre-gyp` requires these and I want to make as few changes as possible

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jtmal-signalfx) [reply .github/workflows/ci.yml:16]: I added `12` and `14` to the test matrix but it looks like we'd some work to support these.

Comment (jtmal-signalfx) [reply .github/workflows/ci.yml:16]: Looks like the tests are just flaky? I'll restructure them and check again on Monday

**Review by Rauno56 — APPROVED** [review]
I'm cool if you'd prefer to add 12+ support in another PR. Request another review if not.

**Pull request merged**